pkg/kubelet/kubelet.go

```diff
@@ -2919,6 +2920,62 @@ func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, string, string) {
 	return true, "", ""
 }
 
+// Get pod resize info in json format
+func getPodResizeInfo(podName string) (string, error) {
+	// Check podAllocs info
+	var jsonData []byte
+	var err error
+
+	podAlloc, exists := allocation.PodResizeReq[podName]
+	if !exists {
+		return "", fmt.Errorf("pod %s info nil", podName)
+	}
+
+	if len(podAlloc.InitContainers) > 0 && len(podAlloc.Containers) > 0 {
+		jsonData, err = json.Marshal(podAlloc)
+	} else if len(podAlloc.InitContainers) > 0 && len(podAlloc.Containers) == 0 {
+		// Defines a temporary structure that only contains the InitContainers word
+		type ContainerOnly struct {
+			InitContainers []allocation.ContainerAllocation `json:"InitContainers"`
+		}
+		// Defines a temporary structure，Assign the InitContainers in PodAllocation to it
+		initcontainerOnly := ContainerOnly{
+			InitContainers: podAlloc.InitContainers,
+		}
+		jsonData, err = json.Marshal(initcontainerOnly)
+
+	} else if len(podAlloc.InitContainers) == 0 && len(podAlloc.Containers) > 0 {
+		// Defines a temporary structure that only contains the Containers word
+		type ContainerOnly struct {
+			Containers []allocation.ContainerAllocation `json:"Containers"`
+		}
+		// Defines a temporary structure，Assign the Containers in PodAllocation to it
+		containerOnly := ContainerOnly{
+			Containers: podAlloc.Containers,
+		}
+		jsonData, err = json.Marshal(containerOnly)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("JSON error: %w", err)
+	}
+
+	finalMsg := fmt.Sprintf("Pod resize completed, %s", string(jsonData))
+
+	// Clear resize data after pod resize complete
+	delete(allocation.PodResizeReq, podName)
```

Comment:
I'd suggest to make PodResizeReq a local entity in allocation package and add required public methods instead of accessing it from other packages directly.